Jenkinsfile

```
@@ -56,33 +66,92 @@ def get_daos_packages(String distro) {
     return pkgs + "-" + daos_packages_version(distro)
 }
 
-def component_repos() {
+def pr_repos() {
     return cachedCommitPragma(pragma: 'PR-repos')
 }
 
-String el7_component_repos() {
+String el7_pr_repos() {
     return cachedCommitPragma(pragma: 'PR-repos-el7')
 }
 
-String leap15_component_repos() {
+String leap15_pr_repos() {
     return cachedCommitPragma(pragma: 'PR-repos-leap15')
 }
 
-def daos_repo() {
+String unit_repos() {
+    Map stage_info = parseStageInfo()
+    return unit_repos(stage_info['target'])
+}
+
+String unit_repos(String distro) {
+    if (distro == 'centos7') {
+        return el7_pr_repos() + ' ' + pr_repos()
+    }
+    if (distro == 'leap15') {
+        return leap15_daos_repos() + ' ' + pr_repos()
+    }
+    error 'unit_test_repos not implemented for ' + distro
+}
+
+String daos_repo() {
     if (cachedCommitPragma(pragma: 'RPM-test-version') == '') {
         return "daos@${env.BRANCH_NAME}:${env.BUILD_NUMBER}"
     } else {
         return ""
     }
 }
 
+String hw_distro_target() {
+    if (env.STAGE_NAME.contains('Hardware')) {
+        if (env.STAGE_NAME.contains('Small')) {
+            return hw_distro('small')
+        }
+        if (env.STAGE_NAME.contains('Medium')) {
+            return hw_distro('medium')
+        }
+        if (env.STAGE_NAME.contains('Large')) {
+            return hw_distro('large')
+        }
+    }
+    Map stage_info = parseStageInfo()
+    return stage_info['target']
+}
+
+String daos_repos() {
+    String target = hw_distro_target()
+    return daos_repos(target)
+}
+
 String daos_repos(String distro) {
-    if (distro == "leap15") {
-        return leap15_component_repos() + ' ' + component_repos() +
-               ' ' + daos_repo()
-    } else if (distro == "centos7") {
-        return el7_component_repos() + ' ' + component_repos() +
-               ' ' + daos_repo()
+    if (distro == 'centos7') {
+        return el7_pr_repos() + ' ' + pr_repos() + ' ' + daos_repo()
+    }
+    if (distro == 'leap15') {
+        return leap15_pr_repos() + ' ' + pr_repos() + ' ' + daos_repo()
+    }
+    error 'daos_repos not implemented for ' + distro
```

Also, only if you need to push again, a bit more DRYing:
```suggestion
    return unit_repos(distro) + ' ' + daos_repo()
```

But given the suggestion above about collapsing `unit_repos()` into `pr_repos()`, change the above accordingly.